interface/wx/maskededit.h

```diff
@@ -0,0 +1,935 @@
+    @endcode
+
+    By specification, all methods of wxEditFieldFlags return the
+    wxEditFieldFlags object itself to allow chaining multiple method calls,
+    like in this example:
+
+    @code
+    maskedCtrl->SetFieldFlags(n, wxEditFieldFlags().SetFillChar('_'));
+    @endcode
+
+    @library{wxcore}
+    @category{ctrl}
+
+    @see wxMaskedEdit
+
+    @since 3.1.6
+*/
+class wxEditFieldFlags
+{
+public:
+    /**
+        Constructor.
+        Sets the alignment of this wxEditFieldFlags to @a alignment.
+
+        Also, by default, sets the fill char to ' ' (space) and the
+        padding char to '\0' (null).
+
+        @param alignment
+            @c wxALIGN_LEFT (default) or @c wxALIGN_RIGHT.
+
+    */
+    wxEditFieldFlags(wxAlignment alignment = wxALIGN_LEFT);
+
+    /**
+        Sets the alignment of this wxEditFieldFlags to @a alignment.
+
+        @param alignment
+            @c wxALIGN_LEFT (default) or @c wxALIGN_RIGHT.
+
+    */
+    wxEditFieldFlags& SetAlignment(wxAlignment alignment);
+
+    /**
+        Sets the fill char of this wxEditFieldFlags to @a fillchar.
+
+        The fill char is used to fill the empty (i.e. not typed yet)
+        positions in the field for showing them. Typically it is a space.
+
+        @param fillchar
+            Any printable character.
+
+    */
+    wxEditFieldFlags& SetFillChar(wxChar fillchar);
+
+    /**
+        Sets the padding char of this wxEditFieldFlags to @a paddingchar.
+
+        The padding char is used to fill the empty (i.e. not typed yet)
+        positions in the field for the return of wxMaskedEdit::GetPlainValue()
+        and wxMaskedEdit::GetFieldValue().
+
+        Default is '\0' (null) which is interpreted as "don't fill"
+
+        @param paddingchar
+            Any printable character or '\0'.
+
+    */
+    wxEditFieldFlags& SetPaddingChar(wxChar paddingchar);
+
+    /**
+        Returns the alignment used in this object.
+
+        @see SetAlignment()
+    */
+    wxAlignment GetAlignment();
+
+    /**
+        Returns the fill char used in this object.
+
+        @see SetFillChar()
+    */
+    wxChar GetFillChar();
+
+    /**
+        Returns the padding char used in this object.
+
+        @see SetPaddingChar()
+    */
+    wxChar GetPaddingChar();
+};
+
+/** @addtogroup group_maskededit_predef
+/**@{*/
+/**
+    The prototype of function used to test a field of a masked edit control.
+
+    The function accepts as parameters a pointer to the instance of the object
+    that calls this function (your wxMaskedEditText or wxMaskedEditCombo), the
+    field for which this function applies and a pointer to an object containing
+    needed information for this function.
+
+    An example of a currently defined function is wxMaskedRangeCheck().
+
+    @library{wxcore}
+    @category{ctrl}
+
+    @see wxMaskedEdit::SetFieldFunction()
+
+    @since 3.1.6
+
+*/
+typedef long wxMaskedFieldFunc(const wxMaskedEdit*, size_t, void*);
```

Comment:
Why are we using callback functions rather than events for validation here?

Reply:
@vadz ,
Is there any other control that uses events for validation?
I am not really familiar with them, hence the question.

I guess I jjust need to see how its done in other controls to properly implement this.

Reply:
@vadz , 
Just to add - or own wxValidator class doesn't use events - they use  virtual function.